src/loaders.py

```diff
@@ -15,43 +14,101 @@
 from .utils import get_target_size
 
 
-class FixedSizeSampler(Sampler):
-    def __init__(self, images_data, *,  sample_size=None, shuffle=True, **kwargs):
-        self.images_data = images_data.sample(frac=1, random_state=SEED) if shuffle else images_data
+class BaseSampler(Sampler):
+    def __init__(self, images_data, *args, sample_size=None, shuffle=True,
+                 even_class_sampling=False, annotations=None, **kwargs):
+        self.images_data = self._prepare_data(images_data)
         self.sample_size = len(self.images_data) if sample_size is None else min(sample_size, len(self.images_data))
+        self.shuffle = shuffle
+        self.even_class_sampling = even_class_sampling
+        self.annotations = annotations
+        assert self.even_class_sampling is False or self.annotations is not None,\
+            'Annotations are required for class sampling'
+
+        if self.even_class_sampling:
+            self.class_labels = self.annotations['LabelName'].unique()
+            self.num_classes = len(self.class_labels)
+            division = self.sample_size / float(self.num_classes)
+            self.samples_per_class = [round(division * (i + 1)) - round(division * i) for i in range(self.num_classes)]
+
+            self.class_indices = {}
+            for class_label in self.class_labels:
+                imgIds = self.annotations[self.annotations['LabelName'] == class_label]['ImageID'].unique()
+                indices = self.images_data[self.images_data['ImageID'].isin(imgIds)].index.tolist()
+                self.class_indices[class_label] = indices
+            for key, value in self.class_indices.items():
+                print("For class {} there are {} images".format(key, len(value)))
 
     def __iter__(self):
-        indices = random.sample(range(len(self.images_data)), self.sample_size)
-        return iter(indices)
+        return iter(self._get_indices(self._get_sample()))
 
     def __len__(self):
         return self.sample_size
 
+    def _get_sample(self):
+        if not self.even_class_sampling:
+            return np.random.choice(len(self.images_data), self.sample_size, replace=False)
 
```

Comment:
@taraspiotr I think it is cleaner to have with if else rather than just if. What do you think?